### PR TITLE
Root/continuos integration ci

### PR DESCRIPTION
**Pull request by AlefrankM**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,70 +12,71 @@
...
     "test:staged": "jest --bail --findRelatedTests --watchAll=false --passWithNoTests",
+    "format:check": "prettier --write \"src/**/*.ts\"",
     "typeorm": "ts-node -P ./tsconfig.json $(yarn bin typeorm)",
     "migration:create": "yarn run typeorm migration:create --",
     "migration:generate": "yarn run typeorm migration:generate -- -d ./src/config/migrations.config.ts",
     "migration:run": "yarn run typeorm migration:run -d ./src/config/migrations.config.ts",
     "migration:revert": "yarn run typeorm migration:revert -d ./src/config/migrations.config.ts"
   },
   "dependencies": {
-    "@commitlint/config-conventional": "^17.1.0",
```
Comment (mmiranda-stackit) [inline package.json]: is there any particular reason why not using the caret on versioning? This way we won't get patches and small fixes backward compatible with the version we are using.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  env: {
+    node: true,
+    jest: true,
+  },
+  ignorePatterns: ['.eslintrc.js'],
+  rules: {
+    '@typescript-eslint/interface-name-prefix': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
```
Comment (mmiranda-stackit) [inline .eslintrc.js]: I think that we should try to type all 'any' findings instead of just ignoring them. This way we will remember the tech debt we have on our plate :)

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+name: Continuous Integration
```
Comment (mmiranda-stackit) [inline .github/workflows/continuous-integration.yml:1]: Looking very good

File: package.json (hunk 1 above)
Comment (AlefrankM) [reply package.json]: Believe me, you don't wanna have carets. These could break the app if they change their behavior in the next version and it's horrible.

Comment (mmiranda-stackit) [reply package.json]: I believe you, but I don't agree 100% what you are saying. Whereas is true what you say, sometimes you cannot control third party and or open source projects, they could potentially 'break' your tests by a bad release, you can always (and we do) use a lock file, which tells us which version we use at a certain point, which makes easier to hunt down the problem and control the versioning (this is the ideal).

However, by pinning dependencies, you are 'blocking' breaking changes, you are also missing the chance to get fixes, minor changes, or worst yet, security fixes.

There is another thing you are not seeing when it comes to pin dependencies, which is sub-dependencies. What I mean with this is, that even though you are pinning you main dependency version, other packages might use the same package you are, which will lead to have multiple versions of the same package, which at the same time might break your app anyways, or in the better scenario generate a lot of noise in you dependencies tree. 

In a nutshell, what I'm saying is that pinning does not mean is good, or is bad, but it depends on the packages you are pinning. This shouldn't be a rule of thumb when it comes to all packages, but be very careful on what packages are we deciding to pin (there is a huge difference of pinning a version of express than a ui component a friend of mine wrote ;))

**Review by mmiranda-stackit — APPROVED** [review]
lgtm
